src/annotator/util/clean-context.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Export clean versions of standard functions and objects.
+ *
+ * This module exports "clean" versions of various standard functions that may
+ * have been modified in the parent browsing context that the client is
+ * injected into. Code should always use the clean versions from this module,
+ * and never the potentially modified versions from the parent context.
+ *
+ * @module util/clean-context
+ */
+
+'use strict';
+
+// Append an invisible <iframe> to the parent document in order to get a clean
+// browsing context to extract clean objects and functions from.
+var iframe = document.createElement('iframe');
+iframe.style.display = 'none';
+document.body.appendChild(iframe);
```

Comment:
On my shiny new Mac I measured the cost of creating an iframe as being ~6-13ms, so probably up ~100ms on a mobile device. It would be nice not to do this for the vast majority of pages that are not going to need it.

I've asked around whether there is a cheaper way, but an approach I'm pretty sure will work fine for feature detection is just to check whether the stringified function (via `fn.toString()`) contains the text `[native code]`. This is currently a convention but [will most likely be part of the language spec](https://tc39.github.io/Function-prototype-toString-revision/) in future.